### PR TITLE
RUN-3321 RUN-3434 move to net module

### PR DESCRIPTION
**Pull request by datamadic**
@HarsimranSingh @whyn07m3 

This swaps out our custom, native backed `resourceFetcher` with the standard electron [net](https://electron.atom.io/docs/api/net/) module. We still will write the response to disk (we can eval later if keeping in memory is appropriate for all cases, I wish it was, as it is already on disc per Chrome cache rules). 

[tests well](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/59f8bcf6af26a25be2ffc9b9)

### CODE REVIEW

File: src/browser/cached_resource_fetcher.ts [hunk 1]
```diff
@@ -117,17 +125,28 @@ function generateHash(str: string): string {
...
+                    resolve();
+                });
+                binaryWriteStream.once('error', (err: Error) => {
+                    reject(err);
+                });
+                binaryWriteStream.end();
+            });
+        });
+        request.end();
     });
```
Comment (wenjunche) [inline src/browser/cached_resource_fetcher.ts]: no request.on('error', ...) and response.on('error', ...) ?

File: src/browser/cached_resource_fetcher.ts [hunk 2]
```diff
@@ -54,37 +57,42 @@ export function cachedFetch(appUuid: string, fileUrl: string, callback: (error:
...
+    let err: Error;
+
+    try {
+        await prepDownloadLocation(appCacheDir, filePath);
+        await download(fileUrl, filePath);
+    } catch (e) {
+        err = e;
+    }
+
+    callback(err, filePath);
```
Comment (whyn07m3) [inline src/browser/cached_resource_fetcher.ts]: In addition to callbacks, may/should/will this async function return value(s). Maybe it's not needed, just asking...

File: src/browser/cached_resource_fetcher.ts [hunk 3]
```diff
@@ -13,8 +13,8 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
-import {app, resourceFetcher} from 'electron'; // Electron app
-import {stat, mkdir} from 'fs';
+import {app, net} from 'electron'; // Electron app
+import {stat, mkdir, writeFileSync, createWriteStream} from 'fs';
```
Comment (HarsimranSingh) [inline src/browser/cached_resource_fetcher.ts]: Remove writeFileSync if it isn't needed

**Comment by HarsimranSingh** [comment]
You're gonna need to account for this since it's getting merged in first

https://github.com/HadoukenIO/core/pull/220

**Comment by rdepena** [comment]
@datamadic How will this treat 302 response codes, do we need to handle asset acquisition or will the module follow redirects ?

**Comment by datamadic** [comment]
@rdepena looks like redirect works just fine
https://testing-dashboard.openfin.co/#/app/tests/59fc9771af26a25be2ffca2b/edit

**Pull request merged**